### PR TITLE
Update ff version to v88.0

### PR DESCRIPTION
**Pull request by trickypr**
This pull request updates ff to v88, fixing dothq/browser-desktop#108 and dothq/browser-desktop#106. **This pull request will bork the code base if you merge it now**. Before merging the following must be changed
- [x] Builds should work
- [x] Some basic ui bugs (e.g. close button sizes) should be fixed
- [x] Reviewers should be comfortable with there not being any critical bugs.

Note that some changes have been made to the code base. Those are listed in dothq/browser-desktop#111. I am going to be off the internet for another 2 weeks (apart from Fridays) because exams are coming up, so feel free to make changes to this pull request to get it working.

### CODE REVIEW

**Comment by trickypr** [comment]
I believe this pull request matches the quality of nightly and is ready for review.

**Comment by trickypr** [comment]
The identity-icon file doesn't appear to be present in the ff code-base under the same name: https://searchfox.org/mozilla-central/search?q=identity-icon

**Edit:** It now uses `info`

**Comment by trickypr** [comment]
@EnderDev I need a `security-warning` icon to finish this pull request, then it should be ready to open up for review

**Comment by trickypr** [comment]
All of the issues in dothq/browser-desktop#111 have been addressed, I am closing that issue.

**Comment by trickypr** [comment]
Dot browser is now 2 versions behind ff. Firefox v89 has been released

File: l10n/en-US/browser/aboutCertError.ftl [hunk 1]
```diff
@@ -1,5 +0,0 @@
-# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (trickypr) [inline l10n/en-US/browser/aboutCertError.ftl]: Is it fine that the localization files were deleted by melon?

File: patches/browser-base-content-nsContextMenu-js.patch [hunk 2]
```diff
@@ -49,38 +49,24 @@ index b03abc5d09fb1a91795d2b6c5433a6f548881f52..f77cefe7954dd2238e616a7e6f343341
...
--      this.webExtBrowserType === "popup"
-+      this.webExtBrowserType === "popup" ||
-+      gIdentityHandler._hasInvalidPageProxyState()
-     );
-     this.showItem("inspect-separator", showInspect);
+@@ -641,6 +641,13 @@ class nsContextMenu {
+         nsContextMenu.DevToolsShim.isDevToolsUser());
+ 
+     this.showItem("context-viewsource", shouldShow);
++//     Something changed here with ff88. This line was removed. Idk why, so it has been commented out
```
Comment (trickypr) [inline patches/browser-base-content-nsContextMenu-js.patch]: This changed in ff88 and I am not sure what to do about it.

File: l10n/en-US/browser/aboutCertError.ftl (hunk 1 above)
Comment (kierandrewett) [reply l10n/en-US/browser/aboutCertError.ftl]: Yes, I have fixed them and converted them back to patch files.

File: patches/browser-base-content-nsContextMenu-js.patch (hunk 2 above)
Comment (kierandrewett) [reply patches/browser-base-content-nsContextMenu-js.patch]: Find where context-viewinfo is and make sure you add to the disabled attr `gIdentityHandler._hasInvalidPageProxyState()`

Comment (trickypr) [reply patches/browser-base-content-nsContextMenu-js.patch]: The issue is that `context-viewinfo` appears to have been removed or renamed: https://searchfox.org/mozilla-central/search?q=context-viewinfo&path=

Comment (trickypr) [reply patches/browser-base-content-nsContextMenu-js.patch]: Ok, the `View page info` option is gone in ff89, so I am going to assume they removed it in v88.
Screen Shot 2021-06-04 at 3 20 10 pm

Comment (trickypr) [reply patches/browser-base-content-nsContextMenu-js.patch]: I can confirm that this was removed in v88

**Comment by drmonocle-official** [comment]
Hello! 

I just wanna ask how do you made the browser based on Mozilla browser... 
I would like to know and maybe I could help you with Dot!

Please mention me when you answer.
Thank you!

**Comment by trickypr** [comment]
This pull request is being closed in favor of migrating straight to v89. The branch will remain for reference until we merge v89 into nightly

**Pull request closed without merging**